exp/api/v1beta2/rosamachinepool_types.go

```diff
@@ -134,6 +139,43 @@ type RosaMachinePoolAutoScaling struct {
 	MaxReplicas int `json:"maxReplicas,omitempty"`
 }
 
+// RosaMachinePoolUpgradeConfig specifies upgrade management configuration which set MaxUnavailable & MaxSurge number of nodes during upgrade.
+type RosaMachinePoolUpgradeConfig struct {
+	// MaxUnavailable is the maximum number of nodes that can be unavailable during the update.
+	// Value can be an absolute number (ex: 5) or a percentage of desired nodes (ex: 10%).
+	// Absolute number is calculated from percentage by rounding down.
+	//
+	// MaxUnavailable can not be 0 if MaxSurge is 0 and defaults to 0.
+	//
+	// Example: when MaxUnavailable is set to 30%, old nodes can be deleted down to 70% of
+	// desired nodes immediately when the rolling update starts. Once new nodes
+	// are ready, more old nodes be deleted, followed by provisioning new nodes,
+	// ensuring that the total number of nodes available at all times during the
+	// update is at least 70% of desired nodes.
+	//
+	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	// +kubebuilder:default=0
+	// +optional
+	MaxUnavailable string `json:"maxUnavailable,omitempty"`
```

Comment:
please use `intstr.IntOrString` type instead of `string`.
see: https://github.com/kubernetes-sigs/cluster-api/blob/6941e8c66f3de69adfb1f19893afc0f3cf4571e6/api/v1beta1/machinedeployment_types.go#L186